server.js

```diff
@@ -249,59 +151,47 @@ function selectTable(res, backend, min, max) {
       max = '0';
   }
   var params = min + ' ' + max + ' ' + backend;
-  cp.exec(py + ' select_table -p ' + params, function (err, stdout) {
-    if (err) { console.log('seltab ' + err.stack); } else {
+  datastore_api(res, 'select_table', params, function (stdout) {
 	sendJSONResponse(res, stdout);
-    }
   });
 }
 
 function selectAll(res, backend) {
-   cp.exec(py + ' select_all -p ' + backend, function (err, stdout) {
-    if (err) { console.log('selall ' + err.stack); } else {
+   datastore_api(res, 'select_all', backend, function (stdout) {
 	sendJSONResponse(res, stdout);
-    }
   });
 }
 
 function selectRow(res, qid) {
-  cp.exec(py + ' select_row -p ' + qid, function (err, stdout) {
-    if (err) { console.log('selrow' + err.stack); } else {
+  datastore_api(res, 'select_row', qid, function (stdout) {
       sendJSONResponse(res, JSON.stringify(JSON.parse(stdout)));
-    }
   });
 }
 
 function getResults(res, qid) {
-  cp.exec(py + ' get_filename -p ' + qid, function (err, stdout) {
-    if (err) { console.log(' relkeys ' + err.stack); } else {
+  datastore_api(res, 'get_filename', qid, function (stdout) {
       sendJSONResponse(res, stdout);
-    }
   });
 }
 
 function getQueryStatus(res, qid) {
-  cp.exec(py + ' get_query_status -p ' + qid, function (err, stdout) {
-    if (err) { console.log('qs' + err.stack); } else {
+  datastore_api(res, 'get_query_status', qid, function (stdout) {
       sendJSONResponse(res, stdout);
-    }
   });
 }
 
 function getTuples(res, rkey) {
   var params = rkey.userName + ' ' + rkey.programName + ' '
         + rkey.relationName;
-  cp.exec(py + ' get_num_tuples -p ' + params, function (err, stdout) {
-    if (err) { console.log( 'numtuples ' + err.stack); } else {
+  datastore_api(res, 'get_num_tuples', params, function (stdout) {
       sendJSONResponse(res, stdout);
-    }
   });
 }
 
 function getQid() {
   cp.exec(py + ' get_latest_qid', function (err, stdout) {
```

Comment:
Is there a reason why the other functions use datastore_api, but this one does not?